Assets/_Game/Scripts/Inventory/ItemPickup.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ItemPickup : MonoBehaviour
+{
+    public ItemObject item;
+    public int amount = 1;
+
+    private void Awake()
+    {
+        GetComponent<SpriteRenderer>().sprite = item.icon;
+        gameObject.name = item.name;
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision != null && collision.gameObject.tag == "Player")
+        {
+            var player = collision.gameObject.GetComponent<PlayerController>();
+            if(player.inventory.currentSize >= player.inventory.maxSize && !player.inventory.ItemExists(item.name))

```

Comment:
Your items should never be direct handling checking for size or if the item exists already. It should simply notify your inventory that an item should be added and the inventory itself should take care of the logic of adding/making sure it can be added